daemon/api_prompting.go

```diff
@@ -97,6 +100,186 @@ func getUserID(r *http.Request) (uint32, Response) {
+	Metadata interface{}   `json:"metadata,omitempty"`
+}
+
+type UnsupportedValueValue struct {
+	err *prompting_errors.UnsupportedValueError
+}
+
+func (v *UnsupportedValueValue) MarshalJSON() ([]byte, error) {
+	value := make(map[string]invalidFieldValue, 1)
+	value[v.err.Field] = invalidFieldValue{
+		Reason: unsupportedValueReason,
+		Metadata: &struct {
+			Received  interface{} `json:"received-invalid"`
+			Supported []string    `json:"supported"`
+		}{
+			Received:  v.err.Unsupported,
+			Supported: v.err.Supported,
+		},
+	}
+	return json.Marshal(value)
+}
+
+type ParseErrorValue struct {
+	err *prompting_errors.ParseError
+}
+
+func (v *ParseErrorValue) MarshalJSON() ([]byte, error) {
+	value := make(map[string]invalidFieldValue, 1)
+	value[v.err.Field] = invalidFieldValue{
+		Reason: parseErrorReason,
+		Metadata: &struct {
+			Received string `json:"received-invalid"`
+			// TODO: once documentation exists for user-defined fields
+			// DocumentationURL string `json:"documentation"`
+		}{
+			Received: v.err.Invalid,
+			// TODO: once documentation exists for user-defined fields
+			// DocumentationURL: <url>,
+		},
+	}
+	return json.Marshal(value)
+}
+
+type RequestedPathNotMatchedErrorJSON prompting_errors.RequestedPathNotMatchedError
+
+func (v *RequestedPathNotMatchedErrorJSON) MarshalJSON() ([]byte, error) {
+	return json.Marshal(&struct {
+		Received  string `json:"received-pattern"`
+		Requested string `json:"requested-path"`
+	}{
+		Received:  v.Received,
+		Requested: v.Requested,
+	})
+}
+
+type RequestedPermissionsNotMatchedErrorJSON prompting_errors.RequestedPermissionsNotMatchedError
+
+func (v *RequestedPermissionsNotMatchedErrorJSON) MarshalJSON() ([]byte, error) {
+	return json.Marshal(&struct {
+		Received  []string `json:"received-permissions"`
+		Requested []string `json:"requested-permissions"`
+	}{
+		Received:  v.Received,
+		Requested: v.Requested,
+	})
+}
+
+type RuleConflictJSON prompting_errors.RuleConflict
+
+func (r *RuleConflictJSON) MarshalJSON() ([]byte, error) {
+	return json.Marshal(&struct {
+		Permission    string `json:"permission"`
+		Variant       string `json:"variant"`
+		ConflictingID string `json:"conflicting-id"`
+	}{
+		Permission:    r.Permission,
+		Variant:       r.Variant,
+		ConflictingID: r.ConflictingID,
+	})
+}
+
+type RuleConflictErrorJSON prompting_errors.RuleConflictError
+
+func (v *RuleConflictErrorJSON) MarshalJSON() ([]byte, error) {
+	conflictsJSON := make([]RuleConflictJSON, len(v.Conflicts))
+	for i, conflict := range v.Conflicts {
+		conflictsJSON[i] = RuleConflictJSON(conflict)
+	}
+	return json.Marshal(&struct {
+		Conflicts []RuleConflictJSON `json:"conflicts"`
+	}{
+		Conflicts: conflictsJSON,
+	})
+}
+
+func promptingNotRunningError() *apiError {
+	return &apiError{
+		Status:  500, // Internal error
+		Message: "AppArmor Prompting is not running",
+		Kind:    client.ErrorKindAppArmorPromptingNotRunning,
+	}
+}
+
+func promptingError(err error) *apiError {
```

Comment:
since we're doing renames perhaps `newPromptingAPIError` or `asPromptingAPIError` ?